includes/class-wc-payments-account.php

```diff
@@ -1057,23 +1057,28 @@ public function maybe_handle_onboarding() {
 				);
 			}
 
-			$wcpay_connect_param = sanitize_text_field( wp_unslash( $_GET['wcpay-connect'] ) );
-
-			$from_wc_admin_task           = 'WCADMIN_PAYMENT_TASK' === $wcpay_connect_param;
-			$from_wc_admin_incentive_page = false !== strpos( wp_get_referer(), 'path=%2Fwc-pay-welcome-page' );
-			$from_wc_pay_connect_page     = false !== strpos( wp_get_referer(), 'path=%2Fpayments%2Fconnect' );
-			if ( $from_wc_admin_task || $from_wc_pay_connect_page || $from_wc_admin_incentive_page ) {
+			$source = WC_Payments_Onboarding_Service::get_source( (string) wp_get_referer(), $_GET );
+			if ( in_array(
+				$source,
+				[
+					WC_Payments_Onboarding_Service::SOURCE_WCADMIN_PAYMENT_TASK,
+					WC_Payments_Onboarding_Service::SOURCE_WCPAY_CONNECT_PAGE,
+					WC_Payments_Onboarding_Service::SOURCE_WCADMIN_SETTINGS_PAGE,
```

Comment:
We weren't previously checking if it came from the settings page here - just confirming if this is intentional?

Reply:
yes! It will fix the bug when the merchants gets directly to Stripe from the settings page

Reply:
Oh - very nice!